modules/objdetect/src/qrcode.cpp

```diff
@@ -2756,11 +2808,58 @@ bool QRDecode::decodingProcess()
 
     if (errorCode != 0) { return false; }
 
-    for (int i = 0; i < qr_code_data.payload_len; i++)
+    CV_LOG_INFO(NULL, "QR: decoded with .version=" << qr_code_data.version << " .data_type=" << qr_code_data.data_type << " .eci=" << qr_code_data.eci << " .payload_len=" << qr_code_data.payload_len)
+
+    switch (qr_code_data.data_type)
     {
-        result_info += qr_code_data.payload[i];
+        case QUIRC_DATA_TYPE_NUMERIC:
+            if (!checkASCIIcompatible(qr_code_data.payload, qr_code_data.payload_len)) {
+                CV_LOG_INFO(NULL, "QR: DATA_TYPE_NUMERIC payload must be ACSII compatible string");
+                return false;
+            }
+            result_info.assign((const char*)qr_code_data.payload, qr_code_data.payload_len);
+            return true;
+        case QUIRC_DATA_TYPE_ALPHA:
+            if (!checkASCIIcompatible(qr_code_data.payload, qr_code_data.payload_len)) {
+                CV_LOG_INFO(NULL, "QR: DATA_TYPE_ALPHA payload must be ASCII compatible string");
+                return false;
+            }
+            result_info.assign((const char*)qr_code_data.payload, qr_code_data.payload_len);
+            return true;
+        case QUIRC_DATA_TYPE_BYTE:
+            // https://en.wikipedia.org/wiki/Extended_Channel_Interpretation
+            if (qr_code_data.eci == QUIRC_ECI_UTF_8) {
+                CV_LOG_INFO(NULL, "QR: payload ECI is UTF-8");
+                if (!checkUTF8(qr_code_data.payload, qr_code_data.payload_len)) {
+                    CV_LOG_INFO(NULL, "QUIRC_DATA_TYPE_BYTE with UTF-8 ECI must be UTF-8 compatible string");
+                    return false;
+                }
+                result_info.assign((const char*)qr_code_data.payload, qr_code_data.payload_len);
+            } else if (qr_code_data.eci == 25/*ECI_UTF_16BE*/) {
+                CV_LOG_INFO(NULL, "QR: UTF-16BE ECI is not supported");
```

Comment:
I propose to make it CV_LOG_WARING. INFO is not printed in regular builds.

Reply:
We should not spam with that message. QR detector is usually called for each frame.